VERSIONING.MD

```diff
@@ -0,0 +1,28 @@
+# Keptn Versioning
+
+## Compatibility requirements
+
+This codebase marks all artifacts with a version of the format `MAJOR.MINOR.PATCH` or
+`MAJOR.MINOR.PATCH-next.ID`. For any artifact with a stable release, that is its version does not
+end in `-next.ID`, no backwards-incompatible changes will be made unless incrementing the `MINOR`
+version number. Patch releases with urgent cherry-picked bugfixes will be made by incrementing the
+`PATCH` version.
+
+A backwards-incompatible change affects the public API. The public API is any public reachable endpoint
```

Comment:
In my opinion, public API could be a lot more then what is inferred by this document.

E.g.,

* all (http) endpoints of `keptn/distributor` (it's not just a proxy, it also used to have port 10999 in 0.10.0, which got removed in 0.11.2)
* go-utils, kubernetes-utils (this is source code! but it has functions that act as an API that is used in many integrations)
* end-user facing changes in the CLI (e.g., adding a new mandatory field to `keptn trigger delivery`)
* CloudEvents
* Internal changes that still affect the user (e.g, mongodb upgrade from 0.10.0 to 0.11.2)

My statement in this document would be:

> A breaking change could be but is not limited to
> * public REST API
> * Go Code (e.g., go-utils, kubernetes-utils, go-sdk)
> * CLI Commands
> * CloudEvents

Reply:
Thank you for your suggestions. I will adapt the PR!